deploy/charts/firefly/values.yaml

```diff
@@ -0,0 +1,237 @@
+# Templates the firefly.core configuration file of FireFly, and in some cases configures which ports are exposed.
+config:
+  # Enables the FireFly debug port on 6060 and `DEBUG` level logs
+  debugEnabled: false
+
+  # Enables the Admin API port for dynamic configuration
+  adminEnabled: true
+
+  # Puts a fresh FireFly node into the preinit state, allowing an operator to then setup smart contracts, apply database migrations, etc. before re-configuring the node to proceed.
+  # It is _not_ recommended to configure FireFly nodes in a preinit state for non-development scenarios.
+  preInit: false
+
+  # The name of the organization the FireFly node belongs to
+  organizationName: ""
+
+  # The blockchain identity of the organization e.g. the Ethereum account address
+  organizationIdentity: ""
+
+  # The Ethereum address of the pre-deployed FireFly smart contract
+  fireflyContractAddress: "/instances/contractAddress"
+
+  # The Postgres URL (including credentials and DB) for FireFly to for the database plugin
+  postgresUrl: ""
+
+  # Whether or not to apply schema migrations automatically on startup, not recommended for production
+  postgresAutomigrate: false
+
+  # The URL of the HTTPS DataExchange for the node to use for the dataexchange plugin i.e. private messaging, only needed if `dataexchange.enabled` is set to false
+  dataexchangeUrl: ""
+
+  # The API key used to authenticate to HTTPS DataExchange, only needed if `dataexchange.enabled` is set to false
+  dataexchangeAPIKey: ""
```

Comment:
Is this accounted for under `dataexchange.apiKey`?

Reply:
It depends. The chart's approach to FF config templating has three modes:
1. you override all of it w/ your own templating via `config.dataexchangeOverride`
2. you are deploying a DX alongside your FF instance, and so `dataexchange.enabled` is set. Therefore, if the DX was configured to use an API key via `dataexchange.apiKey` the config will be templated to use that.
3. you are not deploying a DX alongside FF via this chart, and have an external DX instead. You can just provide its config via `config.dataexchangeAPIKey` and `config.dataexchangeURL` at that point.

Relevant code: https://github.com/hyperledger/firefly/pull/114/files#diff-d5f35d3bee4948ee3394457bc8206e8aeb5bf8dffd5cffcd06915d0634f43c95R161-R180

Its a bit convoluted, but this was the only way I could think to provide autowiring of the DX config if using the chart to deploy the DX, but still providing simple and advanced methods for configuring it yourself.